### ISSUE TITLE
[Truffle] Multiple assignment error with Constants

### ISSUE THREAD

**Issue by bjfish** [jruby/jruby#2795]
Error
```
$ bin/jruby -X+T -e 'STATUS_YET, STATUS_STARTED, STATUS_TERMINATED = 0, 1, 2; puts STATUS_YET'
-e:1:in `const_missing': uninitialized constant Object::STATUS_YET (NameError)
	from -e:1:in `<main>'
```
Expected
```
$ ruby  -e 'STATUS_YET, STATUS_STARTED, STATUS_TERMINATED = 0, 1, 2; puts STATUS_YET'
0
```

**Comment by bjfish** [comment]
TODO: add getoptlong once this is fixed

**Comment by chrisseaton** [comment]
I'll take a look at this soon - `getoptlong` and `optparse` would be very useful, as lots of small utilities we could be running use them.

**Comment by chrisseaton** [comment]
Fixed in dcb15518f75ca1f6d6f3c2318cbc151387c9fb1e.

Here's how I debugged it.

I used the `-Xtruffle.translator.print_asts=main` option to print the Truffle AST for each top level method.

```
$ jt build truffle run -Xtruffle.translator.print_asts=main ../test.rb
(lots of stuff...)
../test.rb main
  SequenceNode
    body[0] = SetTopLevelBindingNode
    body[1] = TopLevelRaiseHandler
      body = SetMethodDeclarationContext
        child = CatchRetryAsErrorNode
          body = CatchReturnAsErrorNode
            body = CatchNextNode
              body = SequenceNode
                body[0] = TraceNode
                body[1] = AssignmentWrapperNode
                  child = ElidableResultNode
                    required = SequenceNode
                      body[0] = WriteLocalVariableUninitializedNode
                        rhs = IntegerFixnumLiteralNode
                      body[1] = WriteLocalVariableUninitializedNode
                        rhs = IntegerFixnumLiteralNode
                      body[2] = WriteLocalVariableUninitializedNode
                        rhs = IntegerFixnumLiteralNode
                      body[3] = WriteLocalVariableUninitializedNode
                        rhs = ReadLocalVariableUninitializedNode
                      body[4] = WriteLocalVariableUninitializedNode
                        rhs = ReadLocalVariableUninitializedNode
                      body[5] = WriteLocalVariableUninitializedNode
                        rhs = ReadLocalVariableUninitializedNode
                    elidableResult = UninitialisedArrayLiteralNode
                      values[0] = ReadLocalVariableUninitializedNode
                      values[1] = ReadLocalVariableUninitializedNode
                      values[2] = ReadLocalVariableUninitializedNode
                body[2] = TraceNode
                body[3] = RubyCallNode
                  receiver = SelfNode
                  dispatchHead = CallDispatchHeadNode
                    first = UnresolvedDispatchNode
                  respondToMissing = CallDispatchHeadNode
                    first = UnresolvedDispatchNode
                  respondToMissingCast = BooleanCastUninitializedNode
                  arguments[0] = ReadConstantNode
                    receiver = LexicalScopeNode
                    dispatch = DispatchHeadNode
                      first = UnresolvedDispatchNode
```

I can see where it passes the `IntegerFixnumLiteralNode` to `WriteLocalVariableUninitializedNode` nodes - this is doing `temp0 = 0`. Then I can see where it should be doing `CONST = temp0`, but it's actually using a `WriteLocalVariableUninitializedNode` instead of a `WriteConstantNode`. I fixed `translateDummyAssignment`, which I know is part of the multiple assignment translation, to handle constants and now the AST is this:

```
../test.rb main
  SequenceNode
    body[0] = SetTopLevelBindingNode
    body[1] = TopLevelRaiseHandler
      body = SetMethodDeclarationContext
        child = CatchRetryAsErrorNode
          body = CatchReturnAsErrorNode
            body = CatchNextNode
              body = SequenceNode
                body[0] = TraceNode
                body[1] = AssignmentWrapperNode
                  child = ElidableResultNode
                    required = SequenceNode
                      body[0] = WriteLocalVariableUninitializedNode
                        rhs = IntegerFixnumLiteralNode
                      body[1] = WriteLocalVariableUninitializedNode
                        rhs = IntegerFixnumLiteralNode
                      body[2] = WriteLocalVariableUninitializedNode
                        rhs = IntegerFixnumLiteralNode
                      body[3] = WriteConstantNode
                        module = LexicalScopeNode
                        rhs = ReadLocalVariableUninitializedNode
                      body[4] = WriteConstantNode
                        module = LexicalScopeNode
                        rhs = ReadLocalVariableUninitializedNode
                      body[5] = WriteConstantNode
                        module = LexicalScopeNode
                        rhs = ReadLocalVariableUninitializedNode
                    elidableResult = UninitialisedArrayLiteralNode
                      values[0] = ReadLocalVariableUninitializedNode
                      values[1] = ReadLocalVariableUninitializedNode
                      values[2] = ReadLocalVariableUninitializedNode
                body[2] = TraceNode
                body[3] = RubyCallNode
                  receiver = SelfNode
                  dispatchHead = CallDispatchHeadNode
                    first = UnresolvedDispatchNode
                  respondToMissing = CallDispatchHeadNode
                    first = UnresolvedDispatchNode
                  respondToMissingCast = BooleanCastUninitializedNode
                  arguments[0] = ReadConstantNode
                    receiver = LexicalScopeNode
                    dispatch = DispatchHeadNode
                      first = UnresolvedDispatchNode
```
